docs/external-node.md

```diff
@@ -0,0 +1,532 @@
+kubectl config --kubeconfig=antrea-agent.kubeconfig use-context antrea-agent@$CLUSTER_NAME
+# Copy antrea-agent.kubeconfig to the VM
+```
+
+5. Create `antrea-agent.antrea.kubeconfig` file for `antrea-agent` to access
+   the `antrea-controller` API server.
+
+```bash
+# Specify the antrea-controller API server endpoint. Antrea-Controller needs to be exposed via the Node IP or a
+# public IP that is reachable from the VM
+export ANTREA_API_SERVER="https://172.18.0.1:443"
+export ANTREA_CLUSTER_NAME="antrea"
+TOKEN=$(kubectl -n vm-ns get secrets -o jsonpath="{.items[?(@.metadata.annotations['kubernetes\.io/service-account\.name']=='$SERVICE_ACCOUNT')].data.token}"|base64 --decode)
+kubectl config --kubeconfig=antrea-agent.antrea.kubeconfig set-cluster $ANTREA_CLUSTER_NAME --server=$ANTREA_API_SERVER --insecure-skip-tls-verify=true
+kubectl config --kubeconfig=antrea-agent.antrea.kubeconfig set-credentials antrea-agent --token=$TOKEN
+kubectl config --kubeconfig=antrea-agent.antrea.kubeconfig set-context antrea-agent@$ANTREA_CLUSTER_NAME --cluster=$ANTREA_CLUSTER_NAME --user=antrea-agent
+kubectl config --kubeconfig=antrea-agent.antrea.kubeconfig use-context antrea-agent@$ANTREA_CLUSTER_NAME
+# Copy antrea-agent.antrea.kubeconfig to the VM
+```
+
+6. Create an [`ExternalNode` resource](#the-externalnode-resource) for the VM.
+
+## Installation on Linux VM
+
+### Prerequisites
+
+OVS needs to be installed on the VM. For more information about OVS installation
+please refer to the [getting-started guide](getting-started.md#open-vswitch).
+
+### Installation
+
+1. Build `antrea-agent` binary in the root of the antrea code tree and copy the
+   `antrea-agent` binary from the `bin` directory to the Linux VM.
+
+```bash
+make docker-bin
+```
+
+2. The `antrea-agent.conf` file specifies agent configuration parameters. Copy
+   the [agent configuration file](../build/yamls/externalnode/conf/antrea-agent.conf)
+   to the VM and edit the `antrea-agent.conf` file to set `clientConnection`,
+   `antreaClientConnection` and `externalNodeNamespace` with the correct values.
+   Copy `antrea-agent.antrea.kubeconfig` and `antrea-agent.kubeconfig` files to
+   the VM, that were generated in the step 4 and step 5 of
+   [Prerequisites on Kubernetes cluster](#prerequisites-on-kubernetes-cluster).
+
+```bash
+AGENT_NAMESPACE="vm-ns"
+AGENT_CONF_PATH="/etc/antrea"
+mkdir -p $AGENT_CONF_PATH
+# Copy antrea-agent kubeconfig files
+cp ./antrea-agent.kubeconfig $AGENT_CONF_PATH
+cp ./antrea-agent.antrea.kubeconfig $AGENT_CONF_PATH
+# Update clientConnection and antreaClientConnection
+sed -i "s|kubeconfig: |kubeconfig: $AGENT_CONF_PATH/|g" antrea-agent.conf
+sed -i "s|#externalNodeNamespace: default|externalNodeNamespace: $AGENT_NAMESPACE|g" antrea-agent.conf
+# Copy antrea-agent configuration file
+cp ./antrea-agent.conf $AGENT_CONF_PATH
+```
+
+3. Create `antrea-agent` service. Below is a sample snippet to start
```

Comment:
I don't see the `NODE_NAME` variable anywhere? I know you can derive it from the host name, but I think it would be better to be explicit in these instructions.

IMO, we should also show the actual `ExternalNode` resource in the previous section

Reply:
> I don't see the NODE_NAME variable anywhere? I know you can derive it from the host name, but I think it would be better to be explicit in these instructions.

We mentioned that `NODE_NAME` variable setting in the section `The ExternalNode resource->Name and Namespace`. Here do you mean we should add this in the installation part?

> IMO, we should also show the actual ExternalNode resource in the previous section
Here do you mean we should show the result of command "kubectl get externalNode" in section `The ExternalNode resource`?

@antoninbas